software/base/src/test/java/org/apache/brooklyn/entity/ExpungingJcloudsLocationLiveTest.java

```diff
@@ -0,0 +1,187 @@
+        assertNull(emptySoftwareProcess.getAttribute(Attributes.JCLOUDS_PROVISIONING_STARTED));
+
+        executeInLimitedTime(new Callable<Void>() {
+            public Void call() {
+                Entities.destroy(app);
+                return null;
+            }
+        }, 1, TimeUnit.SECONDS);
+        assertEquals(app.getAttribute(Attributes.SERVICE_STATE_ACTUAL), Lifecycle.STOPPED);
+        assertEquals(app.getAttribute(Attributes.SERVICE_STATE_EXPECTED).getState(), Lifecycle.STOPPED);
+        assertNull(emptySoftwareProcess.getAttribute(Attributes.JCLOUDS_PROVISIONING_STARTED));
+    }
+
+    @Test(groups = "Integration")
+    public void verifyExpungingByonLocationIsQuick() throws Exception {
+        final VanillaSoftwareProcess entity = app.addChild(EntitySpec.create(VanillaSoftwareProcess.class)
+                .configure(VanillaSoftwareProcess.INSTALL_COMMAND, "echo install")
+                .configure(VanillaSoftwareProcess.LAUNCH_COMMAND, "echo launch")
+                .configure(VanillaSoftwareProcess.CHECK_RUNNING_COMMAND, "echo running"));
+        app.addLocations(ImmutableList.of(mgmt.getLocationFactory().createLocation(TestApplication.LOCALHOST_PROVISIONER_SPEC)));
+
+        EntityManagementUtils.start(app);
+
+        succeedsEventually(ImmutableMap.of("timeout", "4s"), new Callable<Boolean>() {
+            public Boolean call() {
+                assertTrue(entity.sensors().get(Attributes.SERVICE_UP));
+                assertNull(entity.getAttribute(Attributes.JCLOUDS_PROVISIONING_STARTED));
+                return entity.sensors().get(Attributes.SERVICE_UP);
+            }
+        });
+
+        executeInLimitedTime(new Callable<Void>() {
+            public Void call() {
+                Entities.destroy(app);
+                return null;
+            }
+        }, 2, TimeUnit.SECONDS);
+        // Make sure that the entity will be stopped fast. Two seconds at most.
+        assertEquals(app.getAttribute(Attributes.SERVICE_STATE_ACTUAL), Lifecycle.STOPPED);
+        assertEquals(app.getAttribute(Attributes.SERVICE_STATE_EXPECTED).getState(), Lifecycle.STOPPED);
+    }
+
+    public static final String PROVIDER = "aws-ec2";
+    public static final String REGION_NAME = "us-west-2";
+    public static final String LOCATION_SPEC = PROVIDER + (REGION_NAME == null ? "" : ":" + REGION_NAME);
+
+    /**
+     * Verifies the behavior described in
+     * <a href="https://issues.apache.org/jira/browse/BROOKLYN-264">BROOKLYN-264 Stop app while VM still being provisioned: vm is left running when app is expunged</a>
+     * <ul>
+     *     <li>ApplicationResource.launch</li>
+     *     <li>wait a few seconds and EntityResponse.expunge</li>
+     *     <li>assert the image is on the cloud</li>
+     * </ul>
+     */
+    @Test(groups = {"Live"})
+    public void verifyJclousMachineIsExpungedWhenStoppedImmediatelyAfterStart() {
```

Comment:
I have to add a test for one more case. Will see how straight it is to handle it.
When an exception occurs during deployment, in other words the execution goes here https://github.com/apache/brooklyn-server/blob/master/locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java#L1062-L1105
and `destroyOnFailure` is false then the machine will not be expunged.